internal/impl/confluent/serde_goavro.go

```diff
@@ -121,8 +123,22 @@ func (s *schemaRegistryEncoder) getAvroEncoder(ctx context.Context, schema franz
 	}, nil
 }
 
-func (s *schemaRegistryDecoder) getGoAvroDecoder(ctx context.Context, schema franz_sr.Schema) (schemaDecoder, error) {
-	schemaSpec, err := resolveGoAvroReferences(ctx, s.client, s.cfg.avro.mapping, schema)
+func tryExtractCommonSchemaFromAvroBytes(rawUnion bool, specBytes []byte) (any, error) {
+	var as any
+	if err := json.Unmarshal(specBytes, &as); err != nil {
+		return nil, err
+	}
+
+	commonSchema, err := avroAnyToCommonSchema(rawUnion, as)
+	if err != nil {
+		return nil, err
+	}
+
+	return commonSchema.ToAny(), nil
+}
+
+func (s *schemaRegistryDecoder) getGoAvroDecoder(ctx context.Context, aschema franz_sr.Schema) (schemaDecoder, error) {
```

Comment:
It's get is not idiomatic Go - how about `GoAvroDecoderForSchema`?

Reply:
its the same for all the other schema formats, we can change it later if we need to refactor the package.